src/main/java/mops/foren/domain/model/PermissionManager.java

```diff
@@ -42,15 +44,10 @@ public Boolean checkPermission(ForumId id, Permission permission) {
      * @param user       - user that is trying to execute an operation which needs permission
      * @return Boolean
      */
-    public Boolean checkPermission(ForumId id, Permission permission, User author, User user) {
-        Boolean check = false;
-        if (OwnRights.hasPermission(permission)) {
-            check = user.equals(author);
-        }
-        if (!check) {
-            check = checkPermission(id, permission);
-        }
-        return check;
+    public Boolean checkPermission(ForumId id, Permission permission,
+                                   @NonNull User author, @NonNull User user) {
+
+        return user.equals(author) || checkPermission(id, permission);
```

Comment:
You are checking if the user is the author OR if he has the Permission "delete_post", but every user has this permission. This returns true in all cases.

Reply:
Every user has the permission to delete his own post, but if I want to delete a post I am not the author of I still need to get an approval if I have the right because of my role.

Reply:
I agree. But at this point we only have the permission DELETE_POST, we would also need DELETE_FOREIGN_POST for Moderator/Admin.

Reply:
I think this is not necessary because the ID is also delivered?!

Reply:
With our current approach to handling permissions I could not find a way to easily avoid this check. Approved.

Reply:
We can implement this later